daml/Marketplace/Trading.daml

```diff
@@ -245,6 +245,156 @@ template BrokerOrder
           create BrokerTrade with isBuy = isBid, ..
           create DepositTransferRequest with sender = broker, ..
 
+template ClearedOrderRequest
+  with
+    order : ClearedOrder
+  where
+    signatory order.ccp, order.exchParticipant
+    ensure order.qty > 0.0
+
+    controller order.exchange can
+      ClearedOrderRequest_Ack : ContractId ClearedOrder
+        with
+          orderId : Int
+        do create order with orderId = orderId
+
+      ClearedOrderRequest_Reject : ()
+        do return ()
+
+-- TODO: Allow for cancelling pending orders
+template ClearedOrder
+  with
+    ccp : Party
+    exchange : Party
+    exchParticipant : Party
+    pair : IdPair
+    isBid : Bool
+    price : Decimal
+    qty : Decimal
+    status : Text
+    orderId : Int
+  where
+    signatory exchange, exchParticipant
+    ensure qty > 0.0
+
+    key (exchange, orderId) : (Party, Int)
+    maintainer key._1
+
+    controller exchange can
+      ClearedOrder_Fill : Optional (ContractId ClearedOrder)
+        with
+          fillQty : Decimal
+          fillPrice : Decimal
+          counterParty : Party -- TODO Might be unneeded params
+          counterOrderId : Int
+          timestamp : Text
+        do
+          assert $ fillQty > 0.0
```

Comment:
Please change these to `assertMsg`, it can get really tricky to debug with just `assert` since they give almost zero info.